core/templates/dev/head/pages/exploration_player/PlayerServices.js

```diff
@@ -480,18 +482,9 @@ oppia.factory('ExplorationPlayerService', [
       // user is not logged in or has not uploaded a profile picture, or the
       // player is in preview mode.
       getUserProfileImage: function() {
-        var DEFAULT_PROFILE_IMAGE_PATH = (
-          UrlInterpolationService.getStaticImageUrl(
-            '/avatar/user_blue_72px.png'));
-
-        if (_isLoggedIn && !_editorPreviewMode) {
-          return $http.get(
-            '/preferenceshandler/profile_picture'
-          ).then(function(response) {
-            var profilePictureDataUrl = response.data.profile_picture_data_url;
-            return (
-              profilePictureDataUrl ? profilePictureDataUrl :
-              DEFAULT_PROFILE_IMAGE_PATH);
+        if (!_editorPreviewMode) {
+          return UserService.getProfileImageDataUrl().then(function(dataUrl) {
+            return dataUrl;
```

Comment:
Hm. Maybe we should rename the overall method to something like getUserProfileImageAsync() if we are returning a promise?

Also should we change this to something that's retrieved once the service is loaded for the first time, and then fetched locally thereafter?

Reply:
Yes.

Sounds good, it will be nice to move the upload for the profile image to this method too, so ve can invalidate the cache easily.

Reply:
OK, sounds good. Could you do that before this PR goes in?

Reply:
I have moved the method for uploading, but decided not to implement the caching, we reload the page quite a lot and I don't think it would be wise to save the profile image into a cache for now. The problems can arise with Oppia opened on multiple devices, if we change the image in one we would need a way to invalidate in the other place. And also since profile picture is sometimes requested with other data (profile and preferences handler) we would need to solve that.